azurelinuxagent/common/protocol/wire.py

```diff
@@ -939,29 +936,17 @@ def filter_package_list(self, family, ga_manifest, goal_state):
             return allowed_list
 
     def get_gafamily_manifest(self, vmagent_manifest, goal_state):
-        for update_goal_state in [False, True]:
-            try:
-                if update_goal_state:
-                    self.update_goal_state(forced=True)
-                    goal_state = self.get_goal_state()
-
-                self._remove_stale_agent_manifest(
-                    vmagent_manifest.family,
-                    goal_state.incarnation)
+        self._remove_stale_agent_manifest(vmagent_manifest.family, goal_state.incarnation)
```

Comment:
if the retry logic refreshes the goal state, do we need to refresh this incarnation?

Reply:
Discussed offline. After looking at the old implementation, the stale goal state's agent manifest would get removed. The new one's would get removed, but reconstructed immediately after. The new implementation has the same outcome; the stale one's will get deleted, the new one's will remain.